engine/src/main/java/io/graphqlcrud/SQLQueryBuilderVisitor.java

```diff
@@ -263,25 +251,138 @@ public String getSQL() {
     public void visitArgument(Field field, GraphQLFieldDefinition definition, GraphQLObjectType type, Argument arg) {
         VistorContext vctx = this.stack.peek();
 
-        org.jooq.Field<Object> left = field(name(vctx.alias, arg.getName()));
+        String argName = arg.getName();
+        Value<?> argValue = arg.getValue();
+
+        if(argName.equals("page")) {
+            //TODO : Walk through page results
+        } else if(argName.equals("filter")) {
+            LOGGER.debug("Walk through filter results");
+            if(!argValue.getChildren().isEmpty())
+                visitFilterInputs(argValue, vctx, null);
+            else
+                throw new RuntimeException("Cannot fetch filter results on " + argValue);
+        } else {
+            org.jooq.Field<Object> left = field(name(vctx.alias, arg.getName()));
+            Condition condition = visitValueType(argValue, left, "eq");
+            visitCondition(null,condition);
+        }
+    }
 
-        Value<?> v = arg.getValue();
+    public Condition visitStringValue(Value v, org.jooq.Field left, String conditionName) {
         Condition c = null;
-        if (v instanceof StringValue) {
+        if(conditionName.equals("eq"))
             c = left.eq(((StringValue)v).getValue());
-        } else if (v instanceof BooleanValue) {
+        else if(conditionName.equals("ne"))
+            c = left.ne(((StringValue) v).getValue());
+        else if(conditionName.equals("lt"))
+            c = left.lt(((StringValue) v).getValue());
+        else if(conditionName.equals("le"))
+            c = left.le(((StringValue) v).getValue());
+        else if(conditionName.equals("gt"))
+            c = left.gt(((StringValue) v).getValue());
+        else if(conditionName.equals("ge"))
+            c = left.in(((StringValue) v).getValue());
+        else if(conditionName.equals("contains"))
+            c = left.contains(((StringValue) v).getValue());
+        else if(conditionName.equals("startsWith"))
+            c = left.startsWith(((StringValue) v).getValue());
+        else if(conditionName.equals("endsWith"))
+            c = left.endsWith(((StringValue) v).getValue());
+        return c;
+    }
+
+    public Condition visitBooleanValue(Value v, org.jooq.Field left, String conditionName) {
+        Condition c = null;
+        if(conditionName.equals("eq"))
             c = left.eq(((BooleanValue)v).isValue());
-        } else if (v instanceof IntValue) {
+        else if(conditionName.equals("ne"))
+            c = left.ne(((BooleanValue) v).isValue());
+        return c;
+    }
+
+    public Condition visitIntValue(Value v, org.jooq.Field left, String conditionName) {
+        Condition c = null;
+        if(conditionName.equals("eq"))
```

Comment:
Amazing job! I see you have all arguments covered. Is that ready for verification?

Reply:
Have to implement `in` and `between` for `float` and `int` types.